src/transform/src/reduce_elision.rs

```diff
@@ -94,6 +94,30 @@ impl ReduceElision {
                             exprs: vec![a.expr.clone()],
                         },
 
+                        // StringAgg takes nested records of strings and outputs a string
+                        AggregateFunc::StringAgg { .. } => {
```

Comment:
If you don't care about the inner args, then I think you can also fix the other 5 functions that have this problem too (anything that takes an `order_by: Vec<ColumnOrder>`).

Reply:
I am working on those next. @cjubb39 is also working on a subset of them in #8571.

Reply:
You might consider attempting to handle the case where `exprs[0]` is a `MirScalarExpr::Literal`.  This transformation can happen to some queries before getting to this optimization.

Here is a sequence of testdrive commands that in my PR #8571 caused the `ListConcat` to be over a `Literal` rather than a `CallVariadic`.  I imagine something similar might apply to `StringAgg`.
```
statement ok
CREATE TABLE qs (q int not null)

query T
SELECT LIST(SELECT TRUE FROM(SELECT AVG(0) FROM qs))::text;
```

I was only made aware of this case due to some randomized testing Philip did on my PR so you might want to talk to him.

I imagine this isn't blocking because you're making a definite step towards correctness without out -- but I'll leave it up to you / mjibson / whoever to decide